plugins/modules/docker_image_build.py

```diff
@@ -248,35 +323,86 @@ def add_args(self, args):
+                        platforms = parsed
+                    else:
+                        platforms.append(parsed)
+                except ValueError:
+                    platforms.append(platform_param)
+            else:
+                self.fail("Invalid platform format. Expected string, JSON list, or YAML list.")
+        return platforms
+
+    def handle_secrets(self):
+        secrets = self.secret or []
+        temp_files = []
+        for secret in secrets:
+            if 'value' in secret:
+                temp_fd, temp_path = tempfile.mkstemp()
+                with os.fdopen(temp_fd, 'w') as tmp:
+                    tmp.write(str(secret['value']))
+                secret['src'] = temp_path
+                temp_files.append(temp_path)
+
+            elif 'src' in secret and secret['type'] == 'file':
+                if not os.path.isfile(secret['src']):
+                    self.fail("Secret file {0} not found.".format(secret['src']))
+            else:
+                self.fail("Secret must include either a 'value' or a 'src' key.")
+
+        return temp_files
 
     def build_image(self):
-        image = self.client.find_image(self.name, self.tag)
-        results = dict(
-            changed=False,
-            actions=[],
-            image=image or {},
-        )
-
-        if image:
-            if self.rebuild == 'never':
-                return results
-
-        results['changed'] = True
-        if not self.check_mode:
-            args = ['buildx', 'build', '--progress', 'plain']
-            self.add_args(args)
-            args.extend(['--', self.path])
-            rc, stdout, stderr = self.client.call_cli(*args)
-            if rc != 0:
-                self.fail('Building %s:%s failed' % (self.name, self.tag), stdout=to_native(stdout), stderr=to_native(stderr))
-            results['stdout'] = to_native(stdout)
-            results['stderr'] = to_native(stderr)
-            results['image'] = self.client.find_image(self.name, self.tag) or {}
+        temp_files = self.handle_secrets()
+        try:
+            results = dict(changed=False, actions=[], image={})
+            image = self.client.find_image(self.name, self.tag)
+            if image:
+                if self.rebuild == 'never':
+                    results['image'] = image
+                    return results
+            results['changed'] = True
+
+            if not self.check_mode:
+                args = ['buildx', 'build', '--progress', 'plain']
+                self.add_args(args)
+                args.extend(['--', self.path])
+                rc, stdout, stderr = self.client.call_cli(*args)
+
+                if rc != 0:
+                    self.fail('Building %s:%s failed' % (self.name, self.tag), stdout=to_native(stdout), stderr=to_native(stderr))
+
+                results.update(stdout=to_native(stdout), stderr=to_native(stderr), image=self.client.find_image(self.name, self.tag) or {})
```

Comment:
Please keep closer to the original code instead of rewriting everything. Thanks.